scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -111,6 +116,34 @@ class FossId(
 
             return projectName
         }
+
+        /**
+         * This function fetches credentials for [repoUrl] and insert them between the url scheme and the host. If no
+         * matching host is found the in netrc file, the [repoUrl] is returned untouched.
+         */
+        fun getCredentialsFromNetrc(repoUrl: String): String {
+            val matcher = HOST_EXTRACTOR_REGEX.matchEntire(repoUrl)
+            return if (matcher?.groupValues?.size == 4) {
+                val (_, scheme, host, remainder) = matcher.groupValues
```

Comment:
Instead of this, I think it would be nicer to convert to an URI (also see our `String.toUri()` extension function) and then use the URI elements.

Reply:
I thought about it but then considering the URL reconstruction, the result seems cumbersome:
` "${it.scheme}${it.schemeSpecificPart}${it.authority}$scheme${creds.userName}:${String(creds.password)}${it.host}${it.port}${it.path}${it.query}${it.fragment}"`
or
` repoUrl.replaceFirst(it.host, "$scheme${creds.userName}:${String(creds.password)}${it.host}")`

URI are immutable objects so we cannot do `setUrerInfo`. 
I find the regex here particularly readable. Or am I missing something ?

Reply:
Sounds reasonable, I will do it.